### DISCUSSION TITLE
Schema::hasTable - how to avoid?

### DISCUSSION THREAD

**Discussion by klebk** [spatie/laravel-translation-loader#145] [General]
Hi, first of all thank you for a good package!

I'm trying to find if I made misconfiguration but it looks like maybe this is the way its done.

What is the reason for this part in TranslationLoaderManager?
`
if (is_a($model, LanguageLine::class)) {
    if (! Schema::hasTable($model->getTable())) {
        return parent::load($locale, $group, $namespace);
    }
}
`

This part is making foreach translation checking if table exist?

In debugbar I see hundreds of requests like this:
`select * from information_schema.tables where table_schema = '....' and table_name = 'language_lines' and table_type = 'BASE TABLE'`

Thank you for help.
Have a nice day, Klemen

**Comment by freekmurze** [comment]
Hi,

I believe we removed that check in the latest version of the package. Coul you try upgrading?

If it still exists there, feel free to create a PR that removes it.

**Reply by klebk** [reply]
Hi,

thanks! I didn't noticed, it was fixed in last version 7 days ago with try/catch 👍

**Reply by kepe01** [reply]
I am running version 2.7.4 and still getting this error when trying to setup my containers in my Docker, which stops it.
EDIT:
I think it is because it fails in the try-catch, and then in the catch, the hasTable() method is trying to connect to the DB which does not exist yet in our setup.

**Reply by safarovitch** [reply]
Hi. I am using version 2.7.6 and still getting this error. And this is anoying because I get the error when I try to run composer install on server. And now I don't have any choise but to remove this package and it's usages from the project before uploading to server and reverse it after I run mgirations.

**Reply by safarovitch** [reply]
As I see the load function in TranslationLoaderManager.php file of the package is still checking for `Schema::hasTable` inside catch which is a problem if you want the catch part to catch db exceptions. Adding `DB::connection()->getPdo();` as first line in try block resolved the problem for me.
